psclient/src/main/java/com/majeur/psclient/ui/BattleFragment.kt

```diff
@@ -687,6 +754,83 @@ class BattleFragment : BaseFragment(), BattleRoomMessageObserver.UiCallbacks, Vi
                 .start()
     }
 
+    override fun onSetBattleType(type: BattleRoomMessageObserver.BattleType) {
+        this.battleType = type
+        when (type) {
+            BattleRoomMessageObserver.BattleType.REPLAY -> {
+                showReplayControls()
+                enableReplayControls()
+                hidePlayButtonAndShowPause()
+                observedRoomId = "ReplayRoom"
+            }
+            BattleRoomMessageObserver.BattleType.LIVE -> showBattleControls()
+        }
+    }
+
+    override fun goToLatest() {
+        Timber.d("[goToLatest]")
+        postFullScroll()
+    }
+
+    override fun onEndOfReplay() {
+        Timber.d("[onEndOfReplay]")
+        disableReplayControls()
+//        displayDefaultUiControls()
+        homeFragment.onBattleRoomIsCleared()
+    }
+
+    fun displayDefaultUiControls() {
+        Timber.d("[displayDefaultUiControls]")
+        showBattleControls()
+    }
+
+    private fun showReplayControls() {
+        Timber.d("[showReplayControls]")
+        binding.replayControlsContainer.replayControlsContainer.visibility = View.VISIBLE
+
+        binding.extraActionLayout.visibility = GONE
+//        binding.actionContainer.visibility = View.GONE
+        binding.battleDecisionWidget.visibility = GONE
+        enableReplayControls()
+    }
+
+    private fun showBattleControls() {
+        Timber.d("[showBattleControls]")
+        binding.replayControlsContainer.replayControlsContainer.visibility = GONE
+
+//        binding.actionContainer.actionContainer.visibility = VISIBLE
+        binding.extraActionLayout.visibility = VISIBLE
+//        binding.battleDecisionWidget.visibility = VISIBLE
+    }
+
+    private fun disableReplayControls() {
+        Timber.d("[disableReplayControls]")
+        disableImageButton(binding.replayControlsContainer.btnReplayPlay)
+        disableImageButton(binding.replayControlsContainer.btnReplayPause)
+        disableImageButton(binding.replayControlsContainer.btnReplayBack)
+        disableImageButton(binding.replayControlsContainer.btnReplayForward)
+    }
+
+    private fun enableReplayControls() {
+        Timber.d("[enableReplayControls]")
+        enableImageButton(binding.replayControlsContainer.btnReplayPlay)
+        enableImageButton(binding.replayControlsContainer.btnReplayPause)
+        enableImageButton(binding.replayControlsContainer.btnReplayBack)
+        enableImageButton(binding.replayControlsContainer.btnReplayForward)
+    }
+
+    private fun disableImageButton(imageButton: ImageButton) {
+        imageButton.isEnabled = false
+        imageButton.isClickable = false
```

Comment:
isEnabled = false makes a button unclickable already
```suggestion
```

Reply:
Thanks, I didn't realise that.

I removed this and also changed to the 'apply' syntax as you suggested.